### PR TITLE
ci: simplify formatting with julia-actions/julia-format

### PR DESCRIPTION
**Pull request by storopoli**
This simplifies the format CI and delegates everything to [`julia-actions/julia-format`](https://github.com/julia-actions/julia-format). With this workflow, it will automatically post code suggestions to pull requests, based on the formatting rules defined by JuliaFormatter.jl and JuliaFormatter.toml file.

### CODE REVIEW

File: .github/workflows/Format.yml [hunk 1]
```diff
@@ -1,36 +1,12 @@
 name: Format
 
 on:
-  push:
-    branches:
-      - main
```
Comment (rikhuijzer) [inline .github/workflows/Format.yml:6]: Why not? What if someone pushes straight to `main` and then inside a PR someone else is surprised that the formatter fails

File: .github/workflows/Format.yml [hunk 2]
```diff
@@ -1,36 +1,12 @@
...
   pull_request:
   workflow_dispatch:
 
 jobs:
-  docs:
-    name: CheckFormatting
+  code-style:
```
Comment (rikhuijzer) [inline .github/workflows/Format.yml:8]:
```suggestion
  format:
```
maybe?

**Review by rikhuijzer — COMMENTED** [review]
Have you verified that the new approach detects the same formatting problems as the old approach? Does juliaformatter manually go through Markdown files nowadays and does it check for trailing whitespace?

**Comment by storopoli** [comment]
I have changed my mind. I think that what we have works....

**Pull request closed without merging**